cli/src/execDocker/docker.js

```diff
@@ -31,71 +32,88 @@ export async function dockerBuild({
-       *       message: "The command '/bin/sh -c npm ci' returned a non-zero code: 1"
-       *     },
-       *     error: "The command '/bin/sh -c npm ci' returned a non-zero code: 1"
-       *   }
-       * ```
-       */
-      const errorOrErrorMessage =
-        err || // stream error
-        output.find((row) => row?.error)?.error || // build error message
-        (!builtImageId && 'Failed to retrieve generated image ID'); // no image id -> error message
+  try {
+    const imageId = await new Promise((resolve, reject) => {
+      // Handle abort signal
+      if (abortSignal) {
+        abortSignal.addEventListener('abort', () => {
+          buildImageStream.destroy();
+          reject(new Error('Docker build aborted'));
+        });
+      }
 
-      if (errorOrErrorMessage) {
-        const error =
-          errorOrErrorMessage instanceof Error
-            ? errorOrErrorMessage
-            : Error(errorOrErrorMessage);
-        reject(error);
-      } else {
-        resolve(builtImageId);
+      docker.modem.followProgress(buildImageStream, onFinished, onProgress);
+
+      function onFinished(err, output) {
+        /**
+         * expected output format for image id
+         * ```
+         *   {
+         *    aux: {
+         *      ID: 'sha256:e994101ce877e9b42f31f1508e11bbeb8fa5096a1fb2d0c650a6a26797b1906b'
+         *    }
+         *  },
+         * ```
+         */
+        const builtImageId = output?.find((row) => row?.aux?.ID)?.aux?.ID;
+
+        /**
+         * 3 kind of error possible, we want to catch each of them:
+         * - stream error
+         * - build error
+         * - no image id (should not happen)
+         *
+         * expected output format for build error
+         * ```
+         *   {
+         *     errorDetail: {
+         *       code: 1,
+         *       message: "The command '/bin/sh -c npm ci' returned a non-zero code: 1"
+         *     },
+         *     error: "The command '/bin/sh -c npm ci' returned a non-zero code: 1"
+         *   }
+         * ```
+         */
+        const errorOrErrorMessage =
+          err || // stream error
+          output.find((row) => row?.error)?.error || // build error message
+          (!builtImageId && 'Failed to retrieve generated image ID'); // no image id -> error message
+
+        if (errorOrErrorMessage) {
+          const error =
+            errorOrErrorMessage instanceof Error
+              ? errorOrErrorMessage
+              : Error(errorOrErrorMessage);
+          reject(error);
+        } else {
+          resolve(builtImageId);
+        }
       }
-    }
 
-    function onProgress(event) {
-      if (event?.stream) {
-        progressCallback(event.stream);
+      function onProgress(event) {
+        if (event?.stream) {
+          progressCallback(event.stream);
+        }
       }
+    });
+    return imageId;
+  } catch (error) {
+    if (imageId) {
+      await docker.getImage(imageId).remove();
     }
```

Comment:
this never happens `imageId === null` in this context

Reply:
if the built image id is identified, the method is successful, no need to remove the image here